src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs

```diff
@@ -399,17 +401,68 @@ internal long GetLength(ReadOnlySpan<char> path, bool continueOnError = false)
             return EntryExists ? _fileCache.Size : 0;
         }
 
+        internal UnixFileMode GetUnixFileMode(ReadOnlySpan<char> path, bool continueOnError = false)
+            => GetUnixFileMode(handle: null, path, continueOnError);
+
+        internal UnixFileMode GetUnixFileMode(SafeFileHandle handle, bool continueOnError = false)
+            => GetUnixFileMode(handle, handle.Path, continueOnError);
+
+        private UnixFileMode GetUnixFileMode(SafeFileHandle? handle, ReadOnlySpan<char> path, bool continueOnError = false)
+        {
+            EnsureCachesInitialized(handle, path, continueOnError);
+
+            if (!EntryExists)
+                return (UnixFileMode)(-1);
+
+            return (UnixFileMode)(_fileCache.Mode & (int)FileSystem.ValidUnixFileModes);
+        }
+
+        internal void SetUnixFileMode(string path, UnixFileMode mode)
+            => SetUnixFileMode(handle: null, path, mode);
+
+        internal void SetUnixFileMode(SafeFileHandle handle, UnixFileMode mode)
+            => SetUnixFileMode(handle, handle.Path, mode);
+
+        private void SetUnixFileMode(SafeFileHandle? handle, string? path, UnixFileMode mode)
+        {
+            if ((mode & ~FileSystem.ValidUnixFileModes) != 0)
+            {
+                throw new ArgumentException(SR.Arg_InvalidUnixFileMode, "UnixFileMode");
+            }
+
+            EnsureCachesInitialized(handle, path);
+
+            if (!EntryExists)
+                FileSystemInfo.ThrowNotFound(path);
+
+            int rv = handle is not null ? Interop.Sys.FChMod(handle, (int)mode) :
+                                          Interop.Sys.ChMod(path!, (int)mode);
+            Interop.CheckIo(rv, path);
+
+            InvalidateCaches();
+        }
+
+        internal void RefreshCaches(ReadOnlySpan<char> path)
+            => RefreshCaches(handle: null, path);
+
         // Tries to refresh the lstat cache (_fileCache).
         // This method should not throw. Instead, we store the results, and we will throw when the user attempts to access any of the properties when there was a failure
-        internal void RefreshCaches(ReadOnlySpan<char> path)
+        internal void RefreshCaches(SafeFileHandle? handle, ReadOnlySpan<char> path)
```

Comment:
`RefreshCaches` and the changes below to `FileStatus.Unix.cs` is borrowed from https://github.com/dotnet/runtime/pull/60507. No need to review it for this PR.